### PR TITLE
Strip colors from search term

### LINKED ISSUE #4043: GradientChat | Search bug with SlimeFun
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When I'm using /sf open_guide and search while using 
I cannot search because it also gets the color of the message of the player. https://ibb.co/DL6HsL4

### 📑 Reproduction Steps

Just have gradientchat and have a gradient on yourself.

### 💡 Expected Behavior

I'll report here how you can fix it from the developer of the plugin 
image
`    @ParametersAreNonnullByDefault
    public static void openSearch(PlayerProfile profile, String input, SlimefunGuideMode mode, boolean addToHistory) {
        SlimefunGuideImplementation guide = Slimefun.getRegistry().getSlimefunGuide(mode);
        guide.openSearch(profile, ChatColor.stripColor(input), addToHistory);
    }`
    This is (from what he told me) is needed to be fixed so you can search while you have a gradient.

### 📷 Screenshots / Videos

https://ibb.co/DL6HsL4

### 📜 Server Log

It does not matter

### 📂 `/error-reports/` folder

It does not matter

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.19.x

### ⭐ Slimefun version

image

### 🧭 Other plugins

GradientChat
https://www.spigotmc.org/resources/%E2%AD%90gradientplus%E2%AD%90gradient-chat-gradient-name%E2%AD%90best-gradient-plugin.111640/

### PR DESCRIPTION
**Pull request by JustAHuman-xD**
## Description
To allow any servers that have colored chat to still use the slimefun search system

## Proposed changes
- In `SurvivalSlimefunGuide#openSearch` when `searchTerm` is defined I wrapped it with `ChatColor#stripColor`
- In `TestGuideOpening` I changed `testOpenSearch` to `testOpenSearchHistory` to better reflect the test
- In `TestGuideOpening` I created a new `testOpenSearch` method that tests with a normal, and then colored version of the same search. It opens the search for both and asserts that the item registered that should be found by the search is in the upper inventory (where the search would be displayed)

## Related Issues (if applicable)
Resolves #4043 

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [x] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 1]
```diff
@@ -350,7 +350,7 @@ public void openSearch(PlayerProfile profile, String input, boolean addToHistory
         }
 
         ChestMenu menu = new ChestMenu(Slimefun.getLocalization().getMessage(p, "guide.search.inventory").replace("%item%", ChatUtils.crop(ChatColor.WHITE, input)));
-        String searchTerm = input.toLowerCase(Locale.ROOT);
+        String searchTerm = ChatColor.strip(input.toLowerCase(Locale.ROOT));
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:353]:
```suggestion
        String searchTerm = ChatColor.stripColor(input.toLowerCase(Locale.ROOT));
```
I made this PR on my phone, forgot the exact method name lol

**Comment by WalshyDev** [comment]
we've had searching forever and coloured chat is common, how has this not been hit before?

**Comment by JustAHuman-xD** [comment]
> we've had searching forever and coloured chat is common, how has this not been hit before?

no idea

**Review by WalshyDev — DISMISSED** [review]

**Comment by WalshyDev** [comment]
I'd love a test for this, should be easy enough to do.

**Comment by JustAHuman-xD** [comment]
Just made the test, I ran said test with and without my changes and without the change it fails on the colored query but with my change it doesn't, if there's a better way to do the test lmk but this is what I could think of lol

File: src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java [hunk 2]
```diff
@@ -90,8 +93,28 @@ void testOpenItemStack() throws InterruptedException {
...
+        guide.openSearch(profile, normalQuery, false);
+        Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on normal query");
+
+        guide.openSearch(profile, coloredQuery, false);
+        Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on colored query");
+    }
+
+    @Test
+    @DisplayName("Test if the Slimefun Search can be opened from the History")
+    void testOpenSearchHistory() throws InterruptedException {
```
Comment (WalshyDev) [inline src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:117]: this isn't testing history, it's just a basic search term open

I'd rename this back

File: src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java [hunk 3]
```diff
@@ -90,8 +93,28 @@ void testOpenItemStack() throws InterruptedException {
     }
 
     @Test
-    @DisplayName("Test if the Slimefun Search can be opened from the History")
+    @DisplayName("Test if the Slimefun Search works with normal and colored querys")
     void testOpenSearch() throws InterruptedException {
```
Comment (WalshyDev) [inline src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:97]: I like this test, let's name this
```suggestion
    void testOpenSearch_withColoredSearchTerm() throws InterruptedException {
```

File: src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java [hunk 4]
```diff
@@ -90,8 +93,28 @@ void testOpenItemStack() throws InterruptedException {
...
+
+        SlimefunItem testItem = TestUtilities.mockSlimefunItem(plugin, "IRON_ITEM", new CustomItemStack(Material.IRON_INGOT, "iron item"));
+        testItem.register(plugin);
+
+        Player player = server.addPlayer();
+        PlayerProfile profile = TestUtilities.awaitProfile(player);
+        SlimefunGuideImplementation guide = new SurvivalSlimefunGuide(false, false);
+
+        guide.openSearch(profile, normalQuery, false);
+        Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on normal query");
```
Comment (WalshyDev) [inline src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:109]:
```suggestion
        // Assert we can open with a non coloured search term
                Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on normal query");
```

File: src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java [hunk 5]
```diff
@@ -90,8 +93,28 @@ void testOpenItemStack() throws InterruptedException {
...
+
+        guide.openSearch(profile, coloredQuery, false);
+        Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on colored query");
```
Comment (WalshyDev) [inline src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:112]:
```suggestion
        // Assert we can open with a non coloured search term
        Assertions.assertTrue(player.getOpenInventory().getTopInventory().contains(testItem.getItem()), "Failed on colored query");
```

**Review by WalshyDev — APPROVED** [review]

File: src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java (hunk 2 above)
Comment (JustAHuman-xD) [reply src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:117]: Then what does the display name mean?

Comment (WalshyDev) [reply src/test/java/io/github/thebusybiscuit/slimefun4/core/guide/TestGuideOpening.java:117]: Nah the mockito call doesn't do anything for that but I missed that it did a history.add in the playerprofile setup

So yeah we can keep the name

**Review by WalshyDev — APPROVED** [review]

**Review by Sfiguz7 — APPROVED** [review]
Blame Walshy if tests fail

**Pull request merged**